xpra/platform/win32/keyboard.py

```diff
@@ -60,6 +94,16 @@ def init_vars(self):
         #workaround for "fr" keyboards, which use a different key name under X11:
         KEY_TRANSLATIONS[("dead_tilde", 65107,  50)]    = "asciitilde"
         KEY_TRANSLATIONS[("dead_grave", 65104,  55)]    = "grave"
+        self.__x11_layouts_to_win32_hkl = x11_layouts_to_win32_hkl()
+
+    def set_platform_layout(self, layout):
+        hkl = self.__x11_layouts_to_win32_hkl.get(layout)
+        if hkl is None:
+            return 0
+        # https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-activatekeyboardlayout
+        # KLF_SETFORPROCESS|KLF_REORDER = 0x108
+        old_hkl_or_zero_on_failure = ActivateKeyboardLayout(hkl, 0x108)
+        return old_hkl_or_zero_on_failure
```

Comment:
None of the other `set_platform_layout` functions return a value.
If anything, they should all return a boolean. (no point in exposing the "old hkl" to callers - they won't know what to do with it)

Reply:
I removed the return value and just logged the failures because in the "gnome dbus request" scenario, successful/failed response will be determined by a callback and it does not seem good to block the shortkey processing until the callback is invoked.